src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/transpose.hpp

```diff
@@ -67,20 +74,96 @@ void transpose_16NxK(TDST* dst,
+                            T* tmp,
                             const size_t N,
                             const size_t K,
                             const size_t block_size,
                             const size_t dst_stride,
-                            const size_t src_stride) {
+                            const size_t src_stride,
+                            const size_t group_size,
+                            const bool quant_key_bychannel) {
     // will treat as uint32_t transpose
     auto s = reinterpret_cast<uint32_t*>(src);
     auto d = reinterpret_cast<uint32_t*>(dst);
-    transpose_16NxK<uint32_t, ov::element::u32>(d, s, N, K >> 1, block_size, dst_stride, src_stride >> 1);
+    transpose_16NxK<uint32_t, ov::element::u32>(d,
+                                                s,
+                                                nullptr,
+                                                N,
+                                                K >> 1,
+                                                block_size,
+                                                dst_stride,
+                                                src_stride >> 1,
+                                                group_size,
+                                                false);
 }
 #endif
+
+template <typename TDST,
+          ov::element::Type_t SRC_PREC,
+          std::enable_if_t<any_of(SRC_PREC, ov::element::i8, ov::element::u8, ov::element::u4), bool> = true>
+void transpose_16NxK(TDST* dst,
+                     void* src,
+                     TDST* tmp,
+                     const size_t N,
+                     const size_t K,
+                     const size_t block_size,
+                     const size_t dst_stride,
+                     const size_t src_stride,
+                     const size_t group_size,
+                     const bool quant_key_bychannel) {
+    // The layout for per token per head:
+    // |scale(f32)|zeropoint(f32)|quantized feature(u8,idx_1)|quantized feature(u8,idx_2)|...|quantized
+    // feature(u8,idx_S)| The quantized feature will start from 8bytes=sizeof(float)+sizeof(float)
+    auto* s = reinterpret_cast<uint8_t*>(src);
+    constexpr size_t sub_byte_multiplier = get_sub_byte_multiplier(SRC_PREC);
+    constexpr size_t param_count = SRC_PREC == ov::element::i8 ? 1 : 2;
+    auto t = tmp;
+    // if group_size not set, the whole row is used as a group
+    if (quant_key_bychannel) {
+        if constexpr (any_of(SRC_PREC, ov::element::u8, ov::element::u4)) {
+            auto* p_scales = reinterpret_cast<float*>(s);
+            auto* p_zps = p_scales + K;
+            s = s + sizeof(float) * 2 * K;
+            attn_dequant_by_channel_kernel<TDST,
+                                           SRC_PREC>(s, t, N, K, K / sub_byte_multiplier, src_stride, p_scales, p_zps);
+        } else {
+            static_assert(SRC_PREC == ov::element::i8, "i8 doesn't support by-channel quantization");
```

Comment:
It fails for types different than `i8`, but error message suggest that `i8` is not correct.
 ```should the condition be `SRC_PREC != ov::element::i8`?